sys/net/gnrc/network_layer/sixlowpan/gnrc_sixlowpan.c

```diff
@@ -257,7 +257,7 @@ static void _send(gnrc_pktsnip_t *pkt)
 
     /* IP should not send anything here if it is not a 6LoWPAN interface,
      * so we don't need to check for NULL pointers */
-    if (datagram_size <= iface->max_frag_size) {
+    if (gnrc_pkt_len(pkt2->next) <= iface->max_frag_size) {
```

Comment:
Sorry,  I meant the whole packet with `pkt2`, not just the packet snip. The snip `pkt2` is just the netif header. `pkt2->next` is the IPHC header or the IPv6 header respectively. So `gnrc_pkt_len(pkt2->next)` is the length of the whole (compressed) datagram, not just its payload.